client/src/views/Report/SunBurst.js

```diff
@@ -0,0 +1,207 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable react/forbid-prop-types */
+/* eslint-disable react/no-array-index-key */
+import React, { useCallback, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+    PieChart,
+    Pie,
+    Cell,
+    Tooltip,
+} from 'recharts';
+
+import styles from './styles.scss';
+
+const getSunBurstData = (dataObject) => {
+    const centerPieData = [{
+        key: dataObject.key,
+        name: dataObject.name,
+        size: dataObject.size,
+    }];
+
+    const firstLevelData = [];
+    const secondLevelData = [];
+
+    dataObject.children.forEach(({
+        key,
+        name,
+        size,
+        children,
+    }) => {
+        firstLevelData.push({
+            key,
+            name,
+            size,
+        });
+        secondLevelData.push(children.flat());
+    });
+
+    return {
+        centerPieData,
+        firstLevelData,
+        secondLevelData: secondLevelData.flat(),
+    };
+};
+
+const getFilteredData = (newlevel, newKey, dataValue) => {
+    if (newlevel === 'first') {
+        const filteredata = dataValue.children.filter(dvc => dvc.key === newKey);
+        return {
+            centerPieData: [{
+                key: dataValue.key,
+                name: dataValue.name,
+                size: dataValue.size,
+            }],
+            firstLevelData: [{
+                key: filteredata[0].key,
+                name: filteredata[0].name,
+                size: filteredata[0].size,
+            }],
+            secondLevelData: filteredata[0].children,
+        };
+    }
+    if (newlevel === 'second') {
+        const tempData = getSunBurstData(dataValue);
+        return {
+            centerPieData: [],
+            firstLevelData: tempData.firstLevelData.filter(fld => newKey.includes(fld.key)),
+            secondLevelData: tempData.secondLevelData.filter(fld => fld.key === newKey),
+        };
+    }
+    return getSunBurstData(dataValue);
+};
+
+const getCellColor = (dataKey, colorSchemeValue) => {
+    if (dataKey.includes('Primary')) {
+        return colorSchemeValue[0];
+    }
+    return colorSchemeValue[1];
+};
+
+const CustomTooltip = ({ active, payload }) => {
+    if (active) {
```

Comment:
```suggestion
    if (!active) {
        return null
   }
```
It's better to short circuit as early as possible